astropy/io/ascii/cparser.pyx

```diff
@@ -209,9 +210,17 @@ cdef class CParser:
             fast_reader = {}
         elif fast_reader is False: # shouldn't happen
             raise core.ParameterError("fast_reader cannot be False for fast readers")
-        # parallel and use_fast_reader are False by default
-        use_fast_converter = fast_reader.pop('use_fast_converter', False)
+        expchar = fast_reader.pop('fortran_exp', 'E').upper()
```

Comment:
If I'm following along correctly, I think `fortran_exp` is used to specify the exponent character, but otherwise isn't really related to Fortran per se.  Maybe a better name would be `expchar`?